osf/models/node.py

```diff
@@ -2787,30 +2790,33 @@ def remove_node(self, auth, date=None):
         :param datetime date: `datetime.datetime` or `None`
         """
         # TODO: rename "date" param - it's shadowing a global
-        if not self.can_edit(auth):
+        hierarchy = Node.objects.get_children(self, active=True, include_root=True)
+        is_admin = Contributor.objects.filter(node=OuterRef('pk'), admin=True, user=auth.user)
+        if len(hierarchy.annotate(has_admin_perm=Exists(is_admin)).filter(has_admin_perm=False)):
             raise PermissionsError(
-                '{0!r} does not have permission to modify this {1}'.format(auth.user, self.category or 'node')
+                '{0!r} does not have permission to modify this {1}, or a component in its hierarchy.'.format(auth.user, self.category or 'node')
             )
 
-        if Node.objects.get_children(self, active=True).exists():
-            raise NodeStateError('Any child components must be deleted prior to deleting this project.')
```

Comment:
NodeStateError removed here.  Error thrown at API-level instead for earlier versions.

Reply:
It seems like `QuickFilesNode` still raises a `NodeStateError` in its implementation of `remove_node`, though I know that Quick Files have all their own API dealings so they might not get in each other's way... They're so different because a `QuickFilesNode` can't have a hierarchy, so I can see it not being related, just something I noticed and was curious about!

Reply:
I think the QuickFiles NodeStateError is different! It prevents you from deleting the QuickFilesNode entirely.